src/vcpkg/commands.test-features.cpp

```diff
@@ -323,7 +323,7 @@ namespace vcpkg
         msg::println(msgPrecheckBinaryCache);
         binary_cache.precheck(actions_to_check);
 
-        Util::stable_sort(install_plans, [](const auto& left, const auto& right) {
+        Util::stable_sort(install_plans, [](const InstallPlanAction& left, const InstallPlanAction& right) noexcept {
```

Comment:
So you prefer `(const std::pair<FullPackageSpec, InstallPlanAction>& left, const std::pair<FullPackageSpec, InstallPlanAction>& right)` over `(const auto& left, const auto& right)`?

Reply:
I tend to prefer not making lambdas generic when they don't have to be. I think it's both easier to read, because you can look at the body of the lambda and understand what it does without needing to see everywhere it is used, and it confuses debuggers and similar tools less. Contrast this with variables, where I think `auto` is always fine, because variables can never be understood without all the context around them, and that form generally doesn't confuse tools.

Admittedly, it's probably closer to a style choice than a technical one.

I think generic lambdas are fine when something actually uses that genericity. Long enough type names to prefer `auto` probably exist, but  here, `InstallPlanAction` isn't much longer than `auto` and doesn't even introduce an extra newline.